contracts/pool/BlendedPool.sol

```diff
@@ -28,45 +32,75 @@ contract BlendedPool is AbstractPool {
 
     /// @notice withdraws the caller's liquidity assets
     /// @param _amount to be withdrawn
-    function withdraw(uint256 _amount) public override nonReentrant whenProtocolNotPaused {
+    function withdraw(address _beneficiary, uint256 _amount) public override unlocked(msg.sender) nonReentrant whenProtocolNotPaused {
         if (balanceOf(msg.sender) < _amount) revert InsufficientFunds();
-        if (unlockedToWithdraw(msg.sender) < _amount) revert TokensLocked();
         if (principalBalanceAmount < _amount) revert NotEnoughAssets();
 
         _burn(msg.sender, _amount);
 
-        _emitBalanceUpdatedEvent();
-        emit Withdrawal(msg.sender, _amount);
+        emit Withdrawal(msg.sender, _beneficiary, _amount);
+
+        _transferAssets(_beneficiary, _amount);
+    }
+
+    function borrow(address _to, uint256 _amount) public override nonReentrant {
+        super.borrow(_to, _amount);
+    }
 
-        _transferAssets(msg.sender, _amount);
+    function repay(uint256 _amount) public override nonReentrant {
+        super.repay(_amount);
     }
 
     /// @notice Only called by a RegPool when it doesn't have enough Assets
     /// @param _amount the amount requested for compensation
-    function requestAssets(uint256 _amount) external notZero(_amount) nonReentrant onlyPool {
+    function depositToClosedPool(uint256 _amount) external notZero(_amount) nonReentrant onlyPool {
         if (principalBalanceAmount < _amount) revert NotEnoughAssets();
```

Comment:
We can utilize named errors in full and make interaction with protocol better if we specify some data in errors (can also be useful for FE), e.g. 
```suggestion
        if (principalBalanceAmount < _amount) revert NotEnoughAssets(msg.sender, requestedAssetsAmount, availableAssetsAmount);
```